tests/IBTK/rt0_coarsen_01.cpp

```diff
@@ -0,0 +1,194 @@
+        Pointer<VariableContext> ctx = var_db->getContext("context");
+        Pointer<SideVariable<NDIM, double> > u_sc_var = new SideVariable<NDIM, double>("u_sc");
+        const int u_sc_idx = var_db->registerVariableAndContext(u_sc_var, ctx, IntVector<NDIM>(4));
+        Pointer<SideVariable<NDIM, double> > exact_sc_var = new SideVariable<NDIM, double>("exact_sc");
+        const int exact_sc_idx = var_db->registerVariableAndContext(exact_sc_var, ctx, IntVector<NDIM>(4));
+        // TODO u_cc_var is only for plotting (and testing): remove later
+#define DO_PLOT
+#ifdef DO_PLOT
+        Pointer<CellVariable<NDIM, double> > u_cc_var = new CellVariable<NDIM, double>("u_cc", NDIM);
+        const int u_cc_idx = var_db->registerVariableAndContext(u_cc_var, ctx);
+        Pointer<CellVariable<NDIM, double> > exact_cc_var = new CellVariable<NDIM, double>("exact_cc", NDIM);
+        const int exact_cc_idx = var_db->registerVariableAndContext(exact_cc_var, ctx);
+#endif
+
+        gridding_algorithm->makeCoarsestLevel(patch_hierarchy, 0.0);
+        const int tag_buffer = std::numeric_limits<int>::max();
+        int level_number = 0;
+        while ((gridding_algorithm->levelCanBeRefined(level_number)))
+        {
+            gridding_algorithm->makeFinerLevel(patch_hierarchy, 0.0, 0.0, tag_buffer);
+            ++level_number;
+        }
+
+        const int finest_level = patch_hierarchy->getFinestLevelNumber();
+        for (int ln = 0; ln <= finest_level; ++ln)
+        {
+            Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            level->allocatePatchData(u_sc_idx, 0.0);
+            level->allocatePatchData(exact_sc_idx, 0.0);
+#ifdef DO_PLOT
+            level->allocatePatchData(u_cc_idx, 0.0);
+            level->allocatePatchData(exact_cc_idx, 0.0);
+#endif
+        }
+
+        HierarchyMathOps hier_math_ops("hier_math_ops", patch_hierarchy);
+
+        {
+            muParserCartGridFunction u_fcn("u", app_initializer->getComponentDatabase("u"), grid_geometry);
+            u_fcn.setDataOnPatchHierarchy(u_sc_idx, u_sc_var, patch_hierarchy, 0.0);
+            u_fcn.setDataOnPatchHierarchy(exact_sc_idx, exact_sc_var, patch_hierarchy, 0.0);
+        }
+        Pointer<VisItDataWriter<NDIM> > visit_writer = app_initializer->getVisItDataWriter();
+
+        // The rest is just book-keeping, this is the actual test:
+        {
+            Pointer<PatchLevel<NDIM> > level_0 = patch_hierarchy->getPatchLevel(0);
+            Pointer<PatchLevel<NDIM> > level_1 = patch_hierarchy->getPatchLevel(1);
+
+            // there should only be one patch on each patch level
+            Pointer<SideData<NDIM, double> > u_fc_0_data = level_0->getPatch(0)->getPatchData(u_sc_idx);
+            Pointer<SideData<NDIM, double> > u_fc_1_data = level_1->getPatch(0)->getPatchData(u_sc_idx);
+            const Box<NDIM> patch_box_0 = level_0->getPatch(0)->getBox();
```

Comment:
This will break in parallel (I think). Should we assert that it is running only on one processor?